### PR TITLE
Optimise activity feed rendering

### PR DESCRIPTION
**Pull request by vyvil**
PR to move formatting of activity feed section lists from render, applying formatting only when feed data is changed

### CODE REVIEW

File: src/components/ActivityFeed/ActivityFeed.js [hunk 1]
```diff
@@ -138,38 +138,64 @@ type State = {
...
     };
   }
 
+  componentDidMount() {
+    this.generateFeedSections();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const { tabs = [], feedData = [] } = this.props;
+    if ((tabs.length && !isEqual(tabs, prevProps.tabs))
```
Comment (poocart) [inline src/components/ActivityFeed/ActivityFeed.js]: What if comparing without length check? Should be fine and would look nicer. Or are there any other concerns?

File: src/components/Tabs/Tabs.js [hunk 2]
```diff
@@ -298,10 +300,16 @@ export default class Tabs extends React.Component<Props, State> {
...
             <TabItem
+              activeOpacity={1}
               active={isActive}
-              onPress={() => this.setState({
-                activeTab: id,
-              }, onPress)}
+              onPress={() => {
+                if (onTabChange) onTabChange(true);
+                this.setState({ activeTab: id });
+                requestAnimationFrame(() => {
```
Comment (poocart) [inline src/components/Tabs/Tabs.js]: Maybe this needs to be called after state (active tab ID) has changed or it's ok?

File: src/components/ActivityFeed/ActivityFeed.js (hunk 1 above)
Comment (vyvil) [reply src/components/ActivityFeed/ActivityFeed.js]: Property `tabs` is optional. Hence, if it's not provided, "if" check will return false immediately, without a need to call `isEqual`

File: src/components/Tabs/Tabs.js (hunk 2 above)
Comment (vyvil) [reply src/components/Tabs/Tabs.js]: You mean, as a callback of setState?
All of those actions should happen at the same time - we set an active tab, and we render view, that is related to the active tab

File: src/components/Tabs/Tabs.js [hunk 3]
```diff
@@ -45,6 +45,7 @@ type Props = {
   wrapperStyle?: Object,
   isFloating?: boolean,
   coverColor?: string,
+  onTabChange?: Function,
```
[the remark this answers is not in the document]
Comment (vyvil) [reply src/components/Tabs/Tabs.js]: Okay, I'll update typing in upcoming commit :)

File: src/components/ActivityFeed/ActivityFeed.js [hunk 4]
```diff
@@ -138,38 +138,64 @@ type State = {
...
-function getSortedFeedData(tabs, activeTab, feedData) {
-  if (tabs.length) {
-    const aTab = tabs.find(({ id, data = [] }) => id === activeTab && data.length) || { data: [] };
-    return aTab.data.sort((a, b) => b.createdAt - a.createdAt);
-  } else if (feedData.length) {
-    return feedData.sort((a, b) => b.createdAt - a.createdAt);
-  }
-  return [];
+  tabIsChanging: boolean,
+  formattedFeedData: Object[],
```
[the remark this answers is not in the document]
Comment (vyvil) [reply src/components/ActivityFeed/ActivityFeed.js]: Will update it as well

File: src/components/ActivityFeed/ActivityFeed.js [hunk 5]
```diff
@@ -138,38 +138,64 @@ type State = {
...
+    }
+  }
+
+  generateFeedSections = () => {
+    const { tabs = [], activeTab, feedData = [] } = this.props;
+    const dataSections = [];
+    let feedList = feedData;
+
+    if (tabs.length) {
+      const activeTabInfo = tabs.find(({ id }) => id === activeTab) || { data: [] };
```
[the remark this answers is not in the document]
Comment (vyvil) [reply src/components/ActivityFeed/ActivityFeed.js]: Sure :) Will update

**Comment by vyvil** [comment]
@grilix, applied changes according to your comments. Let me know what you think :)

**Pull request merged**